lib/OpenQA/WebAPI/Auth/OAuth2.pm

```diff
@@ -0,0 +1,69 @@
+# Copyright (C) 2020 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::WebAPI::Auth::OAuth2;
+use Mojo::Base -base;
+
+use Carp 'croak';
+
+sub auth_login {
+    my ($self) = @_;
+    my $config = $self->app->config->{oauth2};
+    croak 'No OAuth2 provider selected' unless my $provider = $config->{provider};
+    croak "Provider $provider not supported" unless $provider eq 'github';
+
+    $self->app->plugin(
+        OAuth2 => {
+            $provider => {
+                key    => $config->{key},
+                secret => $config->{secret},
+            }});
```

Comment:
This looks wrong. Isn't `auth_login`called over and over from a controller? This code should run during startup of the web server, not during a request (or even every request here).

Reply:
Loading the plugin over and over will probably leak something.